techmatters/hrm
# Move contact associations up to model

## Description
This PR moves logic from `contact/contact-data-access.ts`/`create` to `contact/contact.ts`/`createContact`. In particular, all the logic of creating/updating associated records to a new contact is moved, leaving `contact/contact-data-access.ts`/`create` only responsible for creating the contact record. 
The functions called within `contact/contact.ts`/`createContact` needed some slight changes, to allow them receive a pgPromise transaction.

### Checklist
- [n/a] Corresponding issue has been opened

### Verification steps
- Confirm changes in the tests make sense & CI passes.
- Regression tests around new contacts (with csam, referrals, using chat channels to check jobs).
